core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannelManager.scala

```diff
@@ -124,18 +127,30 @@ class TransactionMarkerChannelManager(config: KafkaConfig,
                                       networkClient: NetworkClient,
                                       txnStateManager: TransactionStateManager,
                                       txnMarkerPurgatory: DelayedOperationPurgatory[DelayedTxnMarker],
-                                      time: Time) extends Logging {
+                                      time: Time,
+                                      metrics:Metrics) extends Logging {
+
+  private def addQueueLengthMetric(sensor: Sensor, namePrefix: String): Unit = {
+    sensor.add(metrics.metricName(s"$namePrefix-queue-length", "transaction-marker-channel-metrics"), new Total())
```

Comment:
There are some issues with Coda Hale back then, mainly because its api compatibility were consecutively broken in each new release (they may well be fixed now) and hence when we developed the new clients we added our own metrics package.